### ISSUE TITLE
static_host_map: Only one IP is used if the DNS name resolves to multiple IPs

### ISSUE THREAD

**Issue by SamSirry** [slackhq/nebula#832]
In one use case the lighthouse system has two interfaces (eth0, eth1), one with IPv4 and the other IPv6 address configured.

Some nodes are reachable over IPv4 only and others over IPv6 only.

When these nodes are configured with the lighthouse like so:

```
static_host_map:
  "10.10.0.254": ["maestro.example.net:1234"]
```

(where maestro.example.net gets resolved to both the IPv4 and IPv6 addresses by the configured servers in `resolv.conf`)

These nodes are always attempting to connect to this lighthouse over IPv4 (and some of them are failing).

I've traced the code to determine whether nebula is actually loading the lighthouse's address as a single or multiple value, and eventually it boiled down to this line:

https://github.com/slackhq/nebula/blob/61b784d2bb956803298f7578eada0588a6f6fd50/udp/udp_all.go#L89

which leads to the function `firstFavoriteAddr` in the net package that determines to favor one IP address over the others and returns it:

https://documentation.help/Golang/iprawsock.htm#L55

And apparently this makes this behavior an `issue` here because it is not the expected behavior (that an app must try all IPs when given a DNS name that resolves to multiple values).

Since that I can hardly read golang code, I cannot offer any proposed solutions, but at least I could make this resolution.

Thank you.

**Comment by SamSirry** [comment]
I couldn't find the source for `` in any repo, but found this commit: 

https://github.com/golang/go/commit/16dbf2182cf1f091c22df07d215586dd38fdbb64

which shows how the logic always prefers an IPv4 address (weirdly, since IPv4 is out of date since ~2013).

**Comment by JohnMaguire** [comment]
Hi @SamSirry - since #796 landed, was released, and is working as of v1.7.1, we should now use any IPs returned from the DNS server rather than just one.

Of particular note to your use case is the `static_map.network` flag which determines which types of IP addresses to return. By default, this is set to "ip4" as IPv4 connections are preferred for Lighthouse connections (this is because the connecting host typically does not know its own public IPv4 address as it's most likely behind a NAT - the only way to discover it is to have the Lighthouse detect it.) You may want to set this to "ip6" for IPv6-only hosts, or "ip" to detect both IPv4 and IPv6 addresses.

**Comment by SamSirry** [comment]
Thank you @johnmaguire 
For completeness, here's a link to the docs about the `static_map` option: https://nebula.defined.net/docs/config/static-map/

And for some reason I missed the notification about your answer, until now 😅 Thanks again.
